### ISSUE TITLE
When using camel case, private methods with two leading underscores raise invalid-name

### ISSUE THREAD

**Issue by cwiede** [pylint-dev/pylint#10189]
### Bug description

```python
class MyClass:
    def __privateMethod(self):
        pass
```

### Configuration

```ini
[BASIC]
method-naming-style=camelCase
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
...
a.py:2:4: C0103: Method name "__privateMethod" doesn't conform to camelCase naming style (invalid-name)
...
```

### Expected behavior

I'd expect that the the name __privateMethod should be OK with the camelCase setting.

### Pylint version

```shell
pylint 3.3.3
astroid 3.3.8
Python 3.9.10 (tags/v3.9.10:f2f3f53, Jan 17 2022, 15:14:21) [MSC v.1929 64 bit (AMD64)]
```

### Additional dependencies

```python

```
